### PR TITLE
Bump fuel-core to 0.26

### PR DESCRIPTION
**Pull request by DefiCake**
Summary of changes:

- Updated fuel-core to v0.26.0, which brings a breaking change in the `ApplicationHeaders`.
- Updated the `FuelBlockHeadersLib` sol library to account for the breaking change
- Updated block committer, fuels-rs and fuels-ts sdks to interact with fuel-core v0.26.0
- Enabled legacy_encoding feature on fuels-rs sdk

### CODE REVIEW

File: docker/fuel-core/asd [hunk 1]
```diff
@@ -0,0 +1,419 @@
+Run the Fuel client node locally
```
Comment (Voxelot) [inline docker/fuel-core/asd:1]: --

Comment (DefiCake) [reply docker/fuel-core/asd]: Gone

**Comment by DefiCake** [comment]
Awaiting https://github.com/FuelLabs/fuel-core/pull/1868

**Comment by Voxelot** [comment]
this is fixed in fuel-core 0.26, however we are likely blocked on a new set of sdk releases now

**Comment by DefiCake** [comment]
> https://github.com/FuelLabs/fuel-core/pull/1868 is released as part of fuel-core 0.26, however we are likely blocked on a new set of sdk releases now

I thought new versions would preserve compatibility with old SDKs from now on. Is this not the case @luizstacio ?

**Review by luizstacio — APPROVED** [review]

**Pull request merged**